org.adempiere.base/src/org/compiere/model/MTable.java

```diff
@@ -601,6 +601,58 @@ public PO getPO (ResultSet rs, String trxName)
 		return po;
 	}	//	getPO
 
+	/**
+	 * Get PO Class Instance
+	 * 
+	 * @param  uuID    UUID
+	 * @param  trxName transaction
+	 * @return         PO for Record or null
+	 */
+	public PO getPOByUU(String uuID, String trxName)
+	{
+		String tableName = getTableName();
+
+		PO po = null;
+		IServiceReferenceHolder<IModelFactory> cache = s_modelFactoryCache.get(tableName);
+		if (cache != null)
+		{
+			IModelFactory service = cache.getService();
+			if (service != null)
+			{
+				po = service.getPO(tableName, uuID, trxName);
+				if (po != null)
+					return po;
+			}
+			s_modelFactoryCache.remove(tableName);
+		}
+
+		List<IServiceReferenceHolder<IModelFactory>> factoryList = Service.locator().list(IModelFactory.class).getServiceReferences();
+		if (factoryList != null)
+		{
+			for (IServiceReferenceHolder<IModelFactory> factory : factoryList)
+			{
+				IModelFactory service = factory.getService();
+				if (service != null)
+				{
+					po = service.getPO(tableName, uuID, trxName);
+					if (po != null)
+					{
+						s_modelFactoryCache.put(tableName, factory);
+						break;
+					}
+				}
+			}
+		}
+
+		if (po == null)
+		{
+			// TODO have to check working.
+			po = new GenericPO(tableName, getCtx(), 0, trxName);
+		}
+
+		return po;
+	} // getPOByUUID
+

```

Comment:
this implementation plus the addition of new method to IModelFactory interface is actually pointless - it is just a unnecessary long way to reach the same necessary path of MTable.getPO(0, trxName) > po.loadByUU(uuid, trxName)